core/msd/src/main/rdl/Workload.rdli

```diff
@@ -36,13 +36,29 @@ resource Workloads GET "/workloads/{ip}" (name=getWorkloadsByIP) {
     }
 }
 
-// Api for doing a PUT on Workload for a domain/service
+// Api to perform a dynamic workload PUT operation for a domain and service
 // Workload details are obtained from the service certificate
-resource WorkloadOptions PUT "/domain/{domainName}/service/{serviceName}/workload" (name=putWorkload) {
+resource WorkloadOptions PUT "/domain/{domainName}/service/{serviceName}/workload/dynamic" (name=putDynamicWorkload) {
     DomainName domainName; // name of the domain
     EntityName serviceName; // name of the service
-    WorkloadOptions options; // workload store to be inserted
-    authenticate;
+    WorkloadOptions options; // metadata about the dynamic workload
+    authorize ("update", "{domainName}:service.{serviceName}");
```

Comment:
Updated authenticate -> authorize

Reply:
this doesn't look right. Isn't this operation carried out by the pinger? So why would the service itself have update right over itself? The correct authorization must be done in the code where we check that the principal of the client certificate is {domainName}.{serviceName}